### ISSUE TITLE
[BUG]: Sound doesn't stop when card/note is buried or suspended

### ISSUE THREAD

**Issue by snowtimeglass** [ankidroid/Anki-Android#14494]
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

1. Open a deck
2. "Deck options" -> enable "Don't play automatically"
3. "Replay audio" in a card with one or some audio files
4. "Bury" or "Suspend" in the middle of audio playing

### Expected behaviour

The sound stops when the next card is displayed.

### Actual behaviour

The sound doesn't stop after the next card is displayed.

### Debug info

```text
AnkiDroid Version = 2.16.5 (953b9bd879269910ee962b520da9705336d8bc2d)

Android Version = 11

ProductFlavor = play

Manufacturer = OPPO

Model = CPH2013

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 11; CPH2013 Build/RKQ1.200903.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/116.0.0.0 Mobile Safari/537.36

ACRA UUID = 460ff45d-dade-406e-b1ac-56d9175ebdb5

New schema = true

Scheduler = std3

Crash Reports Enabled = true


DatabaseV2 Enabled = true
```

### (Optional) Anything else you want to share?

N/A

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by meliache** [comment]
I have an issue in the 2.17 alpha2 where the sound of a card keeps playing even after I answer the card (e.g. "good"). I don't bury or suspend it. Didn't see that in the alpha1. Is that related and also fixed now or a separate bug?

**Comment by snowtimeglass** [comment]
What I have confirmed is as follows:

>I have an issue in the 2.17 alpha2 where the sound of a card keeps playing even after I answer the card (e.g. "good"). 

Reproduced in 2.17alpha2 on my device. 

The issue is also produced even after I bury or suspend the card in 2.17alpha2.

However, neither of them is not reproduced in my 2.17alpha1-debug, which was created just after the commit in #14493. I have just checked it. "Show answer", Answering a card (e.g. "Good"), "Bury" and "Suspend", in either case, the sound playing stops.
